### PR TITLE
38 search on bookmarks

### PR DESCRIPTION
**Pull request by Nitish814112**
fixed !! used outletContext
"/"
image

"/movies"

image

"/tvseries"

image

"/bookmark"

image

### CODE REVIEW

File: src/components/Content/Card.jsx [hunk 1]
```diff
@@ -33,30 +43,37 @@ const Card = ({ id, name, bookmark, poster_path, title, release_date, first_air_
...
-        </li>
-        <li className="flex items-center text-xs">
-          <span>{adult ? "PG" : "UG"}</span>
-        </li>
-      </ul>
-      <h3 className="mt-2 line-clamp-1 text-lg tracking-tight">
-        {title?title:name}</h3>
+          <li className="flex flex-col items-center text-white text-xs">
+            <span className="mr-1">
+              {release_date?.slice(0, 4) ||
```
Comment (yashtibrewal) [inline src/components/Content/Card.jsx:57]: We are banning ternary operator for better readibility.
Hence refactor it to a function.

File: src/pages/Bookmarks/index.jsx [hunk 2]
```diff
@@ -77,13 +77,16 @@ function Bookmarks() {
       .finally(() => setLoading(false));
   }, []);
 
-  const filteredMovies = bookmarkedMovies.filter(movie =>
-    movie.title.toLowerCase().includes(searchQuery.toLowerCase()) // Filter movies by title
+  const filteredMovies = bookmarkedMovies.filter(
+    (movie) =>
```
Comment (yashtibrewal) [inline src/pages/Bookmarks/index.jsx:81]: The better way is

```javascript
const isMatch = !searchQuery || 
                (movie.title?.toLowerCase().includes(searchQuery.toLowerCase()) || 
                false);
```

                
Notice these 2 things.
1. the use of `?` operator.
2. every case in a new line.

File: src/pages/Bookmarks/index.jsx [hunk 3]
```diff
@@ -77,13 +77,16 @@ function Bookmarks() {
...
+      !searchQuery || (movie.title && movie.title.toLowerCase().includes(searchQuery.toLowerCase()))
   );
-
-  const filteredTvSeries = bookmarkedTvSeries.filter(tvSeries =>
-    tvSeries.title.toLowerCase().includes(searchQuery.toLowerCase()) // Filter tv series by title
+  
+  const filteredTvSeries = bookmarkedTvSeries.filter(
+    (tvSeries) =>
+      !searchQuery || (tvSeries.name && tvSeries.name.toLowerCase().includes(searchQuery.toLowerCase()))
```
Comment (yashtibrewal) [inline src/pages/Bookmarks/index.jsx:87]: Here too.

File: src/pages/HomePage.jsx [hunk 4]
```diff
@@ -47,71 +50,79 @@ export default function HomePage() {
...
+    return function () {
+      clearTimeout(id);
+    };
+  }, [searchQuery, dispatch]);
+
   useEffect(() => {
     const merged = [
-      ...trendingMovies.map((movie) => ({ ...movie, media_type: MEDIA_TYPE.MOVIES })),
-      ...trendingTVSeries.map((tv) => ({ ...tv, media_type: MEDIA_TYPE.TV_SERIES })),
+      ...trendingMovies.map((movie) => ({
```
Comment (yashtibrewal) [inline src/pages/HomePage.jsx:74]: Refactor the common code to a different function.

File: src/pages/Movies.jsx [hunk 5]
```diff
@@ -1,89 +1,104 @@
-import { useCallback, useEffect, useMemo } from "react";
+import { useCallback, useEffect } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import List from "../components/Content/List";
-import { fetchAllMovieBookmarks, fetchAllMovies } from "../store/Redux/MovieSlice"
+import { fetchAllMovieBookmarks, fetchAllMovies } from "../store/Redux/MovieSlice"; // Import searchMovies action
```
Comment (yashtibrewal) [inline src/pages/Movies.jsx:4]: Keep the comment on new line, or remove this comment.

File: src/pages/Movies.jsx [hunk 6]
```diff
@@ -1,89 +1,104 @@
...
-      <h1 className={styles.headings}>Movies</h1>
-      <div className={styles.content}>
-        {searchedMovies.map((card, index) => (
-          <div key={index}>
-            <List cards={[{ ...card, media_type: 'movie' }]} />
-          </div>
-        ))}
+        <h1 className={styles.headings}>Search Results</h1>
+        <div className={styles.content}>
+          <List cards={processedSearchedMovies} />
```
Comment (yashtibrewal) [inline src/pages/Movies.jsx:80]: Good

**Review by yashtibrewal — CHANGES_REQUESTED** [review]
Please check inline code comments.

File: src/pages/Bookmarks/index.jsx (hunk 2 above)
Comment (Nitish814112) [reply src/pages/Bookmarks/index.jsx:81]: //made utility fucn.  
const filterContentBySearch = (contentList, searchQuery, key) => {
    if (!searchQuery) return contentList;
    const lowerCaseQuery = searchQuery.toLowerCase();
    return contentList.filter(
      (content) =>
        content[key] && content[key].toLowerCase().includes(lowerCaseQuery)
    );
  };

File: src/pages/Bookmarks/index.jsx (hunk 3 above)
Comment (Nitish814112) [reply src/pages/Bookmarks/index.jsx:87]: used utility

**Review by yashtibrewal — CHANGES_REQUESTED** [review]
Please push the changes so I can view it on github itself.

File: src/pages/Movies.jsx (hunk 5 above)
Comment (Nitish814112) [reply src/pages/Movies.jsx:4]: resolved

**Review by yashtibrewal — APPROVED** [review]

**Pull request merged**